tsdb/functions.go

```diff
@@ -1262,39 +1163,18 @@ type topMapOut struct {
 func (t topMapOut) Len() int      { return len(t.points) }
 func (t topMapOut) Swap(i, j int) { t.points[i], t.points[j] = t.points[j], t.points[i] }
 func (t topMapOut) Less(i, j int) bool {
-	sortFloat := func(d1, d2 float64) bool {
-		if d1 != d2 {
-			return d1 > d2
-		}
-		k1, k2 := t.points[i].Time, t.points[j].Time
-		if k1 != k2 {
-			return k1 < k2
-		}
-		return t.lessKey(i, j)
-	}
-
-	sortInt64 := func(d1, d2 int64) bool {
-		if d1 != d2 {
-			return d1 > d2
-		}
-		k1, k2 := t.points[i].Time, t.points[j].Time
-		if k1 != k2 {
-			return k1 < k2
-		}
-		return t.lessKey(i, j)
-	}
-
-	sortUint64 := func(d1, d2 uint64) bool {
-		if d1 != d2 {
-			return d1 > d2
-		}
-		k1, k2 := t.points[i].Time, t.points[j].Time
-		if k1 != k2 {
-			return k1 < k2
-		}
-		return t.lessKey(i, j)
-	}
-	return t.less(i, j, sortFloat, sortInt64, sortUint64)
+	// old C trick makes this code easier to read. Imagine
```

Comment:
so the reason that some of this code was originally written the way it was is that we are going to write `bottom` next, and all the logic is somewhat reversed, but not all based on the interface type.  `interfaceCompare` is now hardcoded to be asc, but `bottom` will need sort desc.  How do we handle that scenario now?